src/toStartCaseStr.ts

```diff
@@ -0,0 +1,7 @@
+export function toStartCaseStr(str: string) {
```

Comment:
Maybe some tests? 😅 
Shouldn't we get this maybe from some small (well tested) tree shakable utility from npm?

Reply:
Added tests! 

I looked around for a minimal approach that fit out needs, and included types.. I found this implementation.

LMK if you think having this in our own codebase is a blocker for this PR?

Reply:
Looks good